### PR TITLE
txHandler: metrics for txpool.Remember

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Translate some `Remember` error codes into counters.

## Test Plan

Added unit test to ensure all metrics but two that hard to trigger (ErrNoSpace, ErrTxPoolFeeError).

### CODE REVIEW

File: data/txHandler.go [hunk 1]
```diff
@@ -227,6 +237,50 @@ func (handler *TxHandler) postProcessReportErrors(err error) {
...
+		switch err.Reason {
+		case ledgercore.TxGroupMalformedErrorReasonExceedMaxSize:
+			transactionMessageTxGroupRememberTxGroupTooLarge.Inc(nil)
+		default:
+			transactionMessageTxGroupRememberGroupIDError.Inc(nil)
+		}
+		return
+	}
+
+	transactionMessageTxGroupRememberEvalError.Inc(nil)
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go:281]: kinda, this is error from `eval.transaction()`

File: data/txHandler.go [hunk 2]
```diff
@@ -227,6 +237,50 @@ func (handler *TxHandler) postProcessReportErrors(err error) {
...
+	// it is possible to call errors.As but it requires additional allocations
+	// instead, unwrap and type assert.
+	underlyingErr := errors.Unwrap(err)
+	if underlyingErr == nil {
+		// something went wrong
+		return
+	}
+
+	switch err := underlyingErr.(type) {
+	case *pools.ErrTxPoolFeeError:
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go:265]: this is different and used when tx pool checks fees vs fee per byte computed it is own fees escalation algorithm

File: data/txHandler.go [hunk 3]
```diff
@@ -57,10 +58,19 @@ var transactionMessagesTxnSigVerificationFailed = metrics.MakeCounter(metrics.Tr
...
 var transactionMessagesBacklogSizeGauge = metrics.MakeGauge(metrics.TransactionMessagesBacklogSize)
 
+var transactionMessageTxGroupRememberPoolMaxCap = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberPoolMaxCap)
+var transactionMessageTxGroupRememberNoPendingEval = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberNoPendingEval)
+var transactionMessageTxGroupRememberNoSpace = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberNoSpace)
+var transactionMessageTxGroupRememberFeeError = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberFeeError)
+var transactionMessageTxGroupRememberTxnDead = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberTxnDead)
+var transactionMessageTxGroupRememberTxGroupTooLarge = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberTxGroupTooLarge)
+var transactionMessageTxGroupRememberGroupIDError = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberGroupIDError)
+var transactionMessageTxGroupRememberEvalError = metrics.MakeCounter(metrics.TransactionMessageTxGroupRememberEvalError)
```
[the remark this answers is not in the document]
Comment (iansuvak) [reply data/txHandler.go:72]: I agree that we should use `metrics.TagCounter` for these related metrics. Thoughts on changing the existing metrics above that were added to `postProcessCheckedTxn` recently  as well?

File: data/txHandler.go [hunk 4]
```diff
@@ -227,6 +237,50 @@ func (handler *TxHandler) postProcessReportErrors(err error) {
...
+		return
+	case *ledgercore.TxGroupMalformedError:
+		switch err.Reason {
+		case ledgercore.TxGroupMalformedErrorReasonExceedMaxSize:
+			transactionMessageTxGroupRememberTxGroupTooLarge.Inc(nil)
+		default:
+			transactionMessageTxGroupRememberGroupIDError.Inc(nil)
+		}
+		return
+	}
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go:279]: added

File: data/pools/errors.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+var ErrPendingQueueReachedMaxCap = errors.New("TransactionPool.checkPendingQueueSize: transaction pool have reached capacity")
+
+// ErrNoPendingBlockEvaluator indicates there is no pending block evaluator to accept a new tx group
+var ErrNoPendingBlockEvaluator = errors.New("TransactionPool.ingest: no pending block evaluator")
+
+// ErrTxPoolFeeError is an error type for txpool fee escalation checks
+type ErrTxPoolFeeError string
+
+func (e *ErrTxPoolFeeError) Error() string {
+	return string(*e)
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/pools/errors.go:37]: done

File: data/txHandler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go]: done

**Review by brianolson — APPROVED** [review]
LGTM; some of these new metrics look really useful

**Pull request merged**